test/TestProjects/Unbranded-TypeSpec/src/Generated/UnbrandedTypeSpecClient.cs

```diff
@@ -0,0 +1,2056 @@
+        private readonly TokenCredential _tokenCredential;
+        private readonly MessagePipeline _pipeline;
+        private readonly Uri _endpoint;
+        private readonly string _apiVersion;
+
+        /// <summary> The ClientDiagnostics is used to provide tracing support for the client library. </summary>
+        internal TelemetrySource ClientDiagnostics { get; }
+
+        /// <summary> The HTTP pipeline for sending and receiving REST requests and responses. </summary>
+        public virtual MessagePipeline Pipeline => _pipeline;
+
+        /// <summary> Initializes a new instance of UnbrandedTypeSpecClient for mocking. </summary>
+        protected UnbrandedTypeSpecClient()
+        {
+        }
+
+        /// <summary> Initializes a new instance of UnbrandedTypeSpecClient. </summary>
+        /// <param name="endpoint"> The Uri to use. </param>
+        /// <param name="credential"> A credential used to authenticate to an Azure Service. </param>
+        /// <exception cref="ArgumentNullException"> <paramref name="endpoint"/> or <paramref name="credential"/> is null. </exception>
+        public UnbrandedTypeSpecClient(Uri endpoint, AzureKeyCredential credential) : this(endpoint, credential, new UnbrandedTypeSpecClientOptions())
+        {
+        }
+
+        /// <summary> Initializes a new instance of UnbrandedTypeSpecClient. </summary>
+        /// <param name="endpoint"> The Uri to use. </param>
+        /// <param name="credential"> A credential used to authenticate to an Azure Service. </param>
+        /// <exception cref="ArgumentNullException"> <paramref name="endpoint"/> or <paramref name="credential"/> is null. </exception>
+        public UnbrandedTypeSpecClient(Uri endpoint, TokenCredential credential) : this(endpoint, credential, new UnbrandedTypeSpecClientOptions())
+        {
+        }
+
+        /// <summary> Initializes a new instance of UnbrandedTypeSpecClient. </summary>
+        /// <param name="endpoint"> The Uri to use. </param>
+        /// <param name="credential"> A credential used to authenticate to an Azure Service. </param>
+        /// <param name="options"> The options for configuring the client. </param>
+        /// <exception cref="ArgumentNullException"> <paramref name="endpoint"/> or <paramref name="credential"/> is null. </exception>
+        public UnbrandedTypeSpecClient(Uri endpoint, AzureKeyCredential credential, UnbrandedTypeSpecClientOptions options)
+        {
+            Argument.AssertNotNull(endpoint, nameof(endpoint));
+            Argument.AssertNotNull(credential, nameof(credential));
+            options ??= new UnbrandedTypeSpecClientOptions();
+
+            ClientDiagnostics = new TelemetrySource(options, true);
+            _keyCredential = credential;
+            _pipeline = HttpPipelineBuilder.Build(options, Array.Empty<HttpPipelinePolicy>(), new HttpPipelinePolicy[] { new AzureKeyCredentialPolicy(_keyCredential, AuthorizationHeader) }, new ResponseClassifier());
```

Comment:
```suggestion
            _pipeline = HttpPipelineBuilder.Build(options, Array.Empty<HttpPipelinePolicy>(), new HttpPipelinePolicy[] { new KeyCredentialPolicy(_keyCredential, AuthorizationHeader) }, new ResponseClassifier());
```

This is the policy I'm working on making public right now, but I have an [internal source file](https://github.com/annelo-msft/Demo/pull/4/files#diff-f3e4662494e3a9dab81a367586ccb5cb6529ba9a7a4570985d585776bd09ad2f) you can dump into the generated code directory in the short term if that's needed.